token/program-2022/src/extension/mod.rs

```diff
@@ -505,13 +505,19 @@ impl<'data, S: BaseState> StateWithExtensionsMut<'data, S> {
         &self,
         new_extension: ExtensionType,
     ) -> Result<Option<usize>, ProgramError> {
-        let current_extensions = self.get_extension_types()?;
-        let needed_tlv_len = ExtensionType::get_total_tlv_len(&current_extensions);
-        let new_needed_tlv_len = needed_tlv_len.saturating_add(new_extension.get_type_len());
+        let mut extensions = self.get_extension_types()?;
+        if !extensions.contains(&new_extension) {
+            extensions.push(new_extension);
+        }
+        let new_needed_tlv_len = ExtensionType::get_total_tlv_len(&extensions);
```

Comment:
`get_total_tlv_len` probably also needs a check to make sure that the return is not equal to `Multisig::LEN - BASE_ACCOUNT_LENGTH - size_of::<AccountType>()`.  Maybe we should just move the multisig check from `get_account_len` to `get_total_tlv_len`. What do you think?

Reply:
Good call, moving the multisig check sounds spot on